unicorn/src/unicorn.cpp

```diff
@@ -0,0 +1,795 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#include "cocaine/unicorn.hpp"
+#include "cocaine/unicorn/value.hpp"
+#include "cocaine/unicorn/handlers.hpp"
+#include "cocaine/zookeeper/handler.hpp"
+#include "cocaine/zookeeper/exception.hpp"
+
+#include <cocaine/context.hpp>
+#include <asio/io_service.hpp>
+#include <cocaine/zookeeper/exception.hpp>
+
+#include <memory>
+
+using namespace cocaine::unicorn;
+
+namespace cocaine {
+namespace service {
+
+namespace {
+/**
+* Converts dynamic_t to zookepers config.
+*/
+zookeeper::cfg_t make_zk_config(const dynamic_t& args) {
+    const auto& cfg = args.as_object();
+    const auto& endpoints_cfg = cfg.at("endpoints", dynamic_t::empty_array).as_array();
+    std::vector<zookeeper::cfg_t::endpoint_t> endpoints;
+    for (size_t i = 0; i < endpoints_cfg.size(); i++) {
+        endpoints.emplace_back(endpoints_cfg[i].as_object().at("host").as_string(), endpoints_cfg[i].as_object().at("port").as_uint());
+    }
+    if (endpoints.empty()) {
+        endpoints.emplace_back("localhost", 2181);
+    }
+    return zookeeper::cfg_t(endpoints, cfg.at("recv_timeout", 1000u).as_uint());
+}
+}
+
+void release_lock(unicorn_service_t* service, const path_t& path) {
+
```

Comment:
Yes, to remove that function :-) I've moved the logic of release to lock_state_t